src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
+~~~~~~~~~~~~~~
+
+This document introduces you to three different development environments, all
+based on vstart. Those are:
+
+- vstart running on your host system
+
+- vstart running in a Docker environment
+
+  * ceph-dev-docker_
+  * ceph-dev_
+
+.. note::
+
+  Independently of which of these environments you will choose, you need to
+  compile Ceph in that environment. If you compiled Ceph on your host system,
+  you would have to recompile it on Docker to be able to switch to a Docker
+  based solution. The same is true vice versa. If you previously used a
+  Docker development environment and compiled Ceph there and you now want to
+  switch to your host system, you will also need to recompile Ceph (or
+  compile Ceph using another separate repository).
+
+Development environment on your host system
+...........................................
+
+- No need to learn or have experience with Docker, jump in right away
+- Limited amount of scripts to support automation (like Ceph compilation)
+- No pre-configured easy-to-start services (Prometheus, Grafana, etc)
+- Limited amount of host operating systems supported, depending on which
+  Ceph version is supposed to be used
+
+Development environment using Docker
+....................................
+
+- Some overhead in learning Docker if you are not used to it yet.
+
+- Both Docker projects provide you with scripts that help you getting started
+  and automate recurring tasks.
+
+- Both Docker environments come with partly pre-configured external services
+  which can be used to attach to or complement Ceph Dashboard features, like
+
+  - Prometheus
+  - Grafana
+  - Node-Exporter
+  - Shibboleth
+  - HAProxy
+
+Many of the Ceph Dashboard developers are SUSE employees. We therefore recommend
+you to use `ceph-dev-docker`, which is based on SUSE Tumbleweed. 
+
+vstart on your host system
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The vstart script is usually called from your `build/` directory like so:
+
+.. code::
+
+  ../src/vstart.sh -n -d
+
+In this case `-n` ensures that a new vstart cluster is created (and not the
+previously created on re-used). `-d` enables debug messages in log files.
+There are several more options to chose from. You can get a list using the
+`--help` argument.
+
+Once vstart is running, you will want to check if the `dashboard` manager
+module is running. This can be done by calling `./bin/ceph mgr services` to
+list all the URLs of enabled services. Only URLs of services which are
+available over HTTP(S) will be listed there. Ceph Dashboard is one of these
+services. It should look similar to the following output:
+
+.. code::
+
+  $ ./bin/ceph mgr services
+  {
+      "dashboard": "https://home:41931/",
+      "restful": "https://home:42931/"
+  }
+
+This also reveals the port on which Ceph Dashboard is running on. By default,
+this environment uses a randomly chosen port for Ceph Dashboard and you need
+to use this command to find out which one it has become.
```

Comment:
Actually, the output of `vstart.sh` also prints out the available dashboard URLs:
```
vstart cluster complete. Use stop.sh to stop. See out/* (e.g. 'tail -f out/????') for debug output.

dashboard urls: https://192.168.178.84:41259, https://192.168.178.84:43259, https://192.168.178.84:45259
  w/ user/pass: admin / admin
restful urls: https://192.168.178.84:42259, https://192.168.178.84:44259, https://192.168.178.84:46259
  w/ user/pass: admin / 598da51f-8cd1-4161-a970-b2944d5ad200
```